Make more tests run on Linux

carusogabriel opened the pull request:
@KalleZ @weltling Can you please review this one? :innocent:

KalleZ commented:
I can try take a look at some of these in the weekend.

You have a lot of whitespace changes in this diff btw

carusogabriel commented:
> You have a lot of whitespace changes in this diff btw

Sorry, I do need to disable that in my IDE :disappointed: 

Here at GitHub, you can pass `?w=1`to the query string to hide them, but I'll avoid to make them.

php-pulls commented:
**Comment on behalf of carusogabriel at php.net:**

Labeling

KalleZ commented:
Ack, the main reason we are so on about it is that the unified diffs sent to php-cvs@ becomes extremely hard to read when mixed.

On topic: I think that some of the tests should be renamed, especially if the previous Windows only now works elsewhere

carusogabriel commented:
@KalleZ Done. After that, I've realized that we've became with duplicated identical tests, so I dropped them as well :+1:

nikic commented:
It would be good to run a trailing whitespace strip on all .phpt. It's a bit more tricky that .c files though because in some cases the whitespace actually is relevant.

carusogabriel commented:
@nikic I have a WIP branch locally for that. I'll return to it and open a PR :)

pmmaga commented:
Some of this tests actually don't make sense on linux. I don't think we gain much just by making more tests run if they aren't objectively testing something. The same can be said about the 32bits commit (9c8b3ae).

Some also have their Linux counterparts, if we just remove the windows check but keep the -win32 variants we may be just testing the same thing twice.

Finally, I'd say that if you are removing the check, you could also remove the `<?php include 'skipif.inc'; ?>` and the entire section

carusogabriel commented:
> Finally, I'd say that if you are removing the check, you could also remove the <?php include 'skipif.inc'; ?> and the entire section

We have those checks to skip the tests if a particular extension isn't loaded.

> Some of this tests actually don't make sense on linux. I don't think we gain much just by making more tests run if they aren't objectively testing something. The same can be said about the 32bits commit (9c8b3ae)

My idea is to unify tests to run in every OS. Someone might have added these sections wrongly, and I'm trying to remove them.

carusogabriel commented:
Squashed and rebased. Also, applied changes suggested by @KalleZ about renaming files with `win32` suffix. Also, as pointed by @pmmaga, some tests are now duplicated, so I've dropped the following ones, keeping the one that works across all OS:

```diff
+ext/standard/tests/file/bug43353.phpt
-ext/standard/tests/file/bug43353-win32.phpt

+ext/sockets/tests/socket_send.phpt
-ext/sockets/tests/socket_send_win32.phpt

+ext/date/tests/date_default_timezone_get-1.phpt
-ext/date/tests/date_default_timezone_get-1-win32.phpt
```

carusogabriel commented:
@KalleZ Any objections on merging this one? :)
